ui/src/dataLoaders/actions/dataLoaders.ts

```diff
@@ -378,78 +384,87 @@ export const createOrUpdateTelegrafConfigAsync = () => async (
-  const {
-    dataLoading: {
-      dataLoaders: {telegrafConfigName, telegrafConfigDescription},
-      steps: {bucket, orgID, bucketID},
-    },
-  } = getState()
-
-  const telegrafRequest: TelegrafRequest = {
-    name: telegrafConfigName,
-    description: telegrafConfigDescription,
-    agent: {collectionInterval: DEFAULT_COLLECTION_INTERVAL},
-    organizationID: orgID,
-    plugins,
-  }
+  try {
+    const {
+      dataLoading: {
+        dataLoaders: {telegrafConfigName, telegrafConfigDescription},
+        steps: {bucket, orgID, bucketID},
+      },
+    } = getState()
 
-  // create telegraf config
-  const tc = await client.telegrafConfigs.create(telegrafRequest)
+    const telegrafRequest: TelegrafRequest = {
+      name: telegrafConfigName,
+      description: telegrafConfigDescription,
+      agent: {collectionInterval: DEFAULT_COLLECTION_INTERVAL},
+      organizationID: orgID,
+      plugins,
+    }
 
-  const permissions = [
-    {
-      action: Permission.ActionEnum.Write,
-      resource: {
-        type: PermissionResource.TypeEnum.Buckets,
-        id: bucketID,
-        orgID,
+    // create telegraf config
+    const tc = await client.telegrafConfigs.create(telegrafRequest)
+
+    const permissions = [
+      {
+        action: Permission.ActionEnum.Write,
+        resource: {
+          type: PermissionResource.TypeEnum.Buckets,
+          id: bucketID,
+          orgID,
+        },
       },
-    },
-    {
-      action: Permission.ActionEnum.Read,
-      resource: {type: PermissionResource.TypeEnum.Telegrafs, id: tc.id, orgID},
-    },
-  ]
+      {
+        action: Permission.ActionEnum.Read,
+        resource: {
+          type: PermissionResource.TypeEnum.Telegrafs,
+          id: tc.id,
+          orgID,
+        },
+      },
+    ]
 
-  const token = {
-    name: `${telegrafConfigName} token`,
-    orgID,
-    description: `WRITE ${bucket} bucket / READ ${telegrafConfigName} telegraf config`,
-    permissions,
-  }
+    const token = {
+      name: `${telegrafConfigName} token`,
+      orgID,
+      description: `WRITE ${bucket} bucket / READ ${telegrafConfigName} telegraf config`,
+      permissions,
+    }
 
-  // create token
-  const createdToken = await createAuthorization(token)
+    // create token
+    const createdToken = await createAuthorization(token)
 
-  // add token to data loader state
-  dispatch(setToken(createdToken.token))
+    // add token to data loader state
+    dispatch(setToken(createdToken.token))
 
-  // add token to authorizations state
-  dispatch(addAuthorization(createdToken))
+    // add token to authorizations state
+    dispatch(addAuthorization(createdToken))
 
-  // create label
-  const tokenLabel = {
-    color: '#FFFFFF',
-    description: `token for telegraf config: ${telegrafConfigName}`,
-    tokenID: createdToken.id,
-  } as ILabelProperties // hack to make compiler work
+    // create label
+    const tokenLabel = {
```

Comment:
What do you think about calling this `properties`, so you can use property assignment shorthand below?

Reply:
can do